Makefile

```diff
@@ -38,21 +33,32 @@ build: fmt
 default: help
 
 # -----------------------------------------------------------------------------
-# Docker-based builds
+# Copy files to docker build folder
 # -----------------------------------------------------------------------------
+.PHONY: copy-docker-files
+copy-docker-files:
+	@mkdir -p build/docker/$(GIT_REPOSITORY_NAME)
+	@cp Makefile LICENSE README.md main.go go.mod go.sum build/docker/$(GIT_REPOSITORY_NAME)
+	@cp -r configuration build/docker/$(GIT_REPOSITORY_NAME)
 
-.PHONY: docker-build
-docker-build: docker-rmi-for-build
-	docker build \
-	    --tag $(DOCKER_IMAGE_NAME) \
-		--tag $(DOCKER_IMAGE_NAME):$(GIT_VERSION) \
-		.
+.PHONY: delete-docker-files
+delete-docker-files:
+	@rm -rf build/docker/$(GIT_REPOSITORY_NAME)
+
+# -----------------------------------------------------------------------------
+# Docker-based build
+# -----------------------------------------------------------------------------
 
-.PHONY: docker-build-development-cache
-docker-build-development-cache: docker-rmi-for-build-development-cache
+.PHONY: docker
```

Comment:
I think a Makefile target of `docker` is a little too vague.

Reply:
...but I can see the logic of `make docker`.   The original concept of `make` was to make a file, such as `make main.o`.

Reply:
So something like `make docker-container` would be in line with the original idea of making "files".   ...but I'm not sure I'm in favor of it.  

My initial use of `make docker-build` was to mimic the `docker XXX` command seen at https://docs.docker.com/engine/reference/commandline/docker.   I was considering make targets such as
- `make docker-build`
- `make docker-run`
- `make docker-rmi`
- etc. 

I am willing to reconsider all of these ideas.   Just searching for the most cohesive set of `make` commands.